src/unix/linux-inotify.c

```diff
@@ -108,6 +108,40 @@ static int init_inotify(uv_loop_t* loop) {
 }
 
 
+int uv__inotify_fork(uv_loop_t* loop, void* old_watchers) {
+  /* open the inotify_fd, and re-arm all the inotify watchers */
+  int err;
+  struct watcher_list* tmp_watcher_list;
+  struct watcher_list* watcher_list;
+  QUEUE* q;
+  uv_fs_event_t* handle;
+  char* path;
+
+  if (old_watchers) {
+    /* We must restore the old watcher list to be able to close items
+     *  out of it.
+     */
+    loop->inotify_watchers = old_watchers;
+
+    err = init_inotify(loop);
+    if (err)
+      return err;
+
+    RB_FOREACH_SAFE(watcher_list, watcher_root, CAST(&old_watchers), tmp_watcher_list) {
+      path = watcher_list->path;
+      QUEUE_FOREACH(q, &watcher_list->watchers) {
+        handle = QUEUE_DATA(q, uv_fs_event_t, watchers);
+        uv_fs_event_stop(handle);
```

Comment:
Hmm, to avoid any issues with the watch number (return from `uv__inotify_watch_add`) being reused, I probably need to stop all the handles (thus clearing the watcher list), saving them to a temporary queue, and only then start them up again.